selection_sort.go

```diff
@@ -0,0 +1,16 @@
+package sort
+
+func selectionSort(arr []int) []int {
+	for i := 0; i < len(arr)-1; i++ {
```

Comment:
Optimization: The selection sort algorithm can be optimized slightly by reducing the number of comparisons. Currently, the inner loop runs from `i` to `len(arr)`, but it could be optimized to run from `i+1` to `len(arr)` since the first `i` elements are already sorted. This change won't affect the correctness but can reduce the number of unnecessary comparisons.